### ISSUE TITLE
Can't run command with envar using `hack/env`

### ISSUE THREAD

**Issue by stevekuznetsov** [openshift/origin#12615]
```
$ OS_DEBUG=true OS_BUILD_ENV_PRESERVE=_output/local/releases/rpms:origin.spec:.tito/ hack/env DEBUG=true make build-rpms
[DEBUG] Using commit HEAD
[DEBUG] Using volume origin-build-ad141c9cacd1b91883aa633877f67792b477bf0e
[DEBUG] Re-using volume origin-build-ad141c9cacd1b91883aa633877f67792b477bf0e
[DEBUG] Creating container: `docker create  --privileged -v /var/run/docker.sock:/var/run/docker.sock -v origin-build-ad141c9cacd1b91883aa633877f67792b477bf0e:/go/src/github.com/openshift/origin -e OS_VERSION_FILE= openshift/origin-release:golang-1.7 DEBUG=true make build-rpms`
[DEBUG] Using container beaf6f9daaa6a3380dece9d48d5bb532e7fa1b89d88c1e30b2b4845cf52dfc57
[DEBUG] Syncing source using `tar` and `docker cp`
[DEBUG] Starting container beaf6f9daaa6a3380dece9d48d5bb532e7fa1b89d88c1e30b2b4845cf52dfc57
Error response from daemon: Container command not found or does not exist.
Error: failed to start containers: beaf6f9daaa6a3380dece9d48d5bb532e7fa1b89d88c1e30b2b4845cf52dfc57
[ERROR] PID 5961: hack/lib/build/environment.sh:87: `docker start "${container}" > /dev/null` exited with status 1.
[INFO] 		Stack Trace: 
[INFO] 		  1: hack/lib/build/environment.sh:87: `docker start "${container}" > /dev/null`
[INFO] 		  2: hack/lib/build/environment.sh:167: os::build::environment::start
[INFO] 		  3: hack/lib/build/environment.sh:189: os::build::environment::withsource
[INFO] 		  4: hack/env:33: os::build::environment::run
[INFO]   Exiting with code 1.
[DEBUG] Stopping container beaf6f9daaa6a3380dece9d48d5bb532e7fa1b89d88c1e30b2b4845cf52dfc57
[DEBUG] Removing container beaf6f9daaa6a3380dece9d48d5bb532e7fa1b89d88c1e30b2b4845cf52dfc57
```

**Comment by smarterclayton** [comment]
Hrm - maybe hack-env should propogate all env vars with a certain prefix.

**Comment by smarterclayton** [comment]
I guess there are others we'd like to propagate.  Slightly weird about sending all of them, in case you have some sort of credentials.

**Comment by stevekuznetsov** [comment]
Fixed in #12633
